### ISSUE TITLE
quarkus-kind extension does not create manifests (kind.yml/.json)

### ISSUE THREAD

**Issue by therealjsie** [quarkusio/quarkus#28078]
### Describe the bug

When using the `quarkus-kind` extension, no manifests are created and the deployment subsequently fails.

I'm using the following command to build the project: 
`gradle build -Dquarkus.container-image.build=true -Dquarkus.kubernetes.deploy=true`

### Expected behavior

I expect the `quarkus-kind` extension to create a `kind.yml` and `kind.json` file.

### Actual behavior

No files are created. The deployment fails.

### How to Reproduce?

[crickets.zip](https://github.com/quarkusio/quarkus/files/9605066/crickets.zip)

Use the following command: `gradle build -Dquarkus.container-image.build=true -Dquarkus.kubernetes.deploy=true`

### Output of `uname -a` or `ver`

Darwin V6624MV3PM 21.6.0 Darwin Kernel Version 21.6.0: Mon Aug 22 20:19:52 PDT 2022; root:xnu-8020.140.49~2/RELEASE_ARM64_T6000 arm64

### Output of `java -version`

openjdk 17.0.4 2022-07-19

### Quarkus version or git rev

2.11.3.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

------------------------------------------------------------ Gradle 7.4.2 ------------------------------------------------------------  Build time:   2022-03-31 15:25:29 UTC Revision:     540473b8118064efcc264694cbcaa4b677f61041  Kotlin:       1.5.31 Groovy:       3.0.9 Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021 JVM:          17.0.4 (Eclipse Adoptium 17.0.4+8) OS:           Mac OS X 12.6 aarch64

**Comment by Sgitario** [comment]
In addition to the details of this issue, I've also found:
1. Looking deeper into the code, no `kind.yaml`, `kind.json` are expected to be created... I think the documentation is wrong, but it needs to be confirmed. 
2. If we fix the previous point 1., still the load of the container image is not working because of the issue described in https://kind.sigs.k8s.io/docs/user/known-issues/#unable-to-pull-images. So, we need to append the `--name` in the `load-image` command.

If the reasoning in point 1. is correct, I don't think we need the steps to provide the decorators, labels and so in the Kind processor, only the last build step (the one to perform the load image). Moreover, to fix point 2., users would need to provide the kind cluster name. 

All this needs to be confirmed by either @iocanel or @aureamunoz .

**Comment by therealjsie** [comment]
@Sgitario: Thanks for your response. I forgot to add the exception that I get when I run the following command. This is what led me to believe that there should be a `kind.yml`.

`gradle build -Dquarkus.container-image.build=true -Dquarkus.kubernetes.deploy=true`

```
> io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.kubernetes.deployment.KubernetesDeployer#deploy threw an exception: java.lang.IllegalStateException: Can't find generated kubernetes manifest: /Users/johannessiebel/git-repos/jsie/crickets/build/kubernetes/kind.yml
        at io.quarkus.kubernetes.deployment.KubernetesDeployer.deploy(KubernetesDeployer.java:273)
        at io.quarkus.kubernetes.deployment.KubernetesDeployer.deploy(KubernetesDeployer.java:134)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:977)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
        at java.base/java.lang.Thread.run(Thread.java:833)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```

**Comment by linvsendian** [comment]
Hi, did this realy solved? I'm tring [2.14.0.CR1](https://github.com/quarkusio/quarkus/milestone/206) but nothing happened,  it does not create the kind.yml. I'm using Maven instead of Gradle

**Comment by Sgitario** [comment]
> Hi, is this realy solved? I'm tring [2.14.0.CR1](https://github.com/quarkusio/quarkus/milestone/206) but nothing happened, it does not create the kind.yml. I'm using Maven instead of Gradle

I've just double-checked and it worked fine for me using 2.14.0.CR1. This is the example I used: https://github.com/Sgitario/quarkus-examples/tree/main/kubernetes-kind and set the Quarkus version via the Maven command:
```
mvn clean install -DskipTests -Dquarkus.version=2.14.0.CR1
```

**Comment by Sgitario** [comment]
> > > Hi, is this realy solved? I'm tring [2.14.0.CR1](https://github.com/quarkusio/quarkus/milestone/206) but nothing happened, it does not create the kind.yml. I'm using Maven instead of Gradle
> > 
> > 
> > I've just double-checked and it worked fine for me using 2.14.0.CR1. This is the example I used: https://github.com/Sgitario/quarkus-examples/tree/main/kubernetes-kind and set the Quarkus version via the Maven command:
> > ```
> > mvn clean install -DskipTests -Dquarkus.version=2.14.0.CR1
> > ```
> 
> I created a new quarkus project to try it again, and nothing happened. This is the full trace: $ mvn clean install -DskipTests -Dquarkus.kubernetes.deploy=true -Dquarkus.kubernetes.namespace=default

This is indeed very weird. The first time I downloaded a Quarkus project from scratch, it worked fine, but then it stopped generating the `kind` resources. And, if I add the `quarkus-kubernetes` extension, it consistently generates the `kind` resources.
In my quarkus example, it always generates the Kind resource, so in the Quarkus integration tests that were added.
Any ideas about what could be happening here @iocanel ? I'm reopening this issue because it's not fully resolved.

**Comment by Sgitario** [comment]
> > > > Hi, is this realy solved? I'm tring [2.14.0.CR1](https://github.com/quarkusio/quarkus/milestone/206) but nothing happened, it does not create the kind.yml. I'm using Maven instead of Gradle
> > > 
> > > 
> > > I've just double-checked and it worked fine for me using 2.14.0.CR1. This is the example I used: https://github.com/Sgitario/quarkus-examples/tree/main/kubernetes-kind and set the Quarkus version via the Maven command:
> > > ```
> > > mvn clean install -DskipTests -Dquarkus.version=2.14.0.CR1
> > > ```
> > 
> > 
> > I created a new quarkus project to try it again, and nothing happened. This is the full trace: $ mvn clean install -DskipTests -Dquarkus.kubernetes.deploy=true -Dquarkus.kubernetes.namespace=default
> 
> This is indeed very weird. The first time I downloaded a Quarkus project from scratch, it worked fine, but then it stopped generating the `kind` resources. And, if I add the `quarkus-kubernetes` extension, it consistently generates the `kind` resources. In my quarkus example, it always generates the Kind resource, so in the Quarkus integration tests that were added. Any ideas about what could be happening here @iocanel ? I'm reopening this issue because it's not fully resolved.

Finally, I found a working solution and provided a fix here: https://github.com/quarkusio/quarkus/pull/29166

**Comment by linvsendian** [comment]
Cool!
